mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -755,14 +793,24 @@ impl MakerSwap {
     }
 
     async fn wait_for_taker_payment(&self) -> Result<(Option<MakerSwapCommand>, Vec<MakerSwapEvent>), String> {
-        let maker_payment_hex = self.r().maker_payment.as_ref().unwrap().tx_hex.0.clone();
-        let msg = SwapMsg::MakerPayment(maker_payment_hex);
+        let payment_data_msg = match self.get_my_payment_data().await {
+            Ok(data) => data,
+            Err(e) => {
+                return Ok((Some(MakerSwapCommand::RefundMakerPayment), vec![
+                    MakerSwapEvent::MakerPaymentDataSendFailed(e.to_string().into()),
+                    MakerSwapEvent::MakerPaymentWaitRefundStarted {
+                        wait_until: self.wait_refund_until(),
+                    },
+                ]));
+            },
+        };
+        let msg = SwapMsg::MakerPayment(payment_data_msg);
         let abort_send_handle =
             broadcast_swap_message_every(self.ctx.clone(), swap_topic(&self.uuid), msg, 600., self.p2p_privkey);
 
         let maker_payment_wait_confirm = self.r().data.started_at + (self.r().data.lock_duration * 2) / 5;
         let f = self.maker_coin.wait_for_confirmations(
-            &self.r().maker_payment.clone().unwrap().tx_hex,
+            &self.r().maker_payment.clone().unwrap().tx_hex_or_hash(),
```

Comment:
To be honest, this looks really weird. What if `TransactionIdentifier::tx_hex` will be not `None`, but the same value as `TransactionIdentifier::tx_hash` for lightning? Then we can pass `TransactionIdentifier::tx_hex` as it is here and in other places.

Making `TransactionIdentifier::tx_hex` optional also leads to huge changes in `taker_swap.rs`

Reply:
> What if TransactionIdentifier::tx_hex will be not None, but the same value as TransactionIdentifier::tx_hash for lightning?

I actually implemented it like this in the beginning but thought it might be confusing. Also these checks https://github.com/KomodoPlatform/atomicDEX-API/blob/8b515c860faf640e04438c694561bdc8b9093ee4/mm2src/mm2_main/src/lp_swap/taker_swap.rs#L1377-L1378
https://github.com/KomodoPlatform/atomicDEX-API/blob/8b515c860faf640e04438c694561bdc8b9093ee4/mm2src/mm2_main/src/lp_swap/taker_swap.rs#L1419 will be more confusing since we will check if `tx_hash != tx_hex`. What do you think?

Reply:
Where is it required to check if `tx_hash != tx_hex`?

Reply:
> Where is it required to check if tx_hash != tx_hex?

Instead of checking if `tx_hex` is some

Reply:
Maybe a method can be added to `SwapOps` called `is_watcher_supported` instead of doing these checks, what do you think @caglaryucekaya @sergeyboyko0791?

Reply:
I like this idea
> Maybe a method can be added to SwapOps called is_watcher_supported instead of doing these checks, what do you think @caglaryucekaya @sergeyboyko0791?

Reply:
Will implement it then. @caglaryucekaya I guess `is_watcher_supported` should be true for `UTXO` and `QTUM` only, is that correct?

Reply:
Yes correct, will start to implement the rest in the next sprint.